src/libcrun/cgroup.c

```diff
@@ -736,7 +738,11 @@ systemd_job_removed (sd_bus_message *m, void *userdata, sd_bus_error *error arg_
     return -1;
 
   if (strcmp (p->path, path) == 0)
-    *p->terminated = 1;
+    {
+      *p->terminated = 1;
+      p->unit = xstrdup(unit);
+      p->result = xstrdup(result);
```

Comment:
do we need to free these two strings once they are used?

Reply:
fixed